src/System.Private.ServiceModel/src/Resources/System.Private.ServiceModel.rd.xml

```diff
@@ -29,6 +29,10 @@
         <!-- ServiceContractAttribute on a type makes it available for Reflection -->
         <Type Name="ServiceContractAttribute" Dynamic="Required All" >
           <AttributeImplies Dynamic="Required All" />
+          <!-- Type in CallbackContract named property requires reflection -->
+          <Method Name=".ctor">
+             <TypeParameter Name="CallbackContract" Dynamic="Required All" />
+          </Method>
         </Type>
         <!-- ExceptionDetail requires serialization for FaultException -->
```

Comment:
Would something like the following work?
```xml
<Type Name="InstanceContext" Dynamic="Required All" >  
    <Method Name=".ctor">  
        <Parameter Name="implementation" Dynamic="Required All" />  
    </Method>  
</Type>  
```
As any callback implementation type is passed to WCF wrapped in an `InstanceContext`, we might be able to get the compiler to generate the metadata based on the `InstanceContext` constructor